ldj-spring-renderer-example/src/main/java/com/infosupport/ldoc/springexample/ampq/SpringAmpqRenderer.java

```diff
@@ -0,0 +1,234 @@
+package com.infosupport.ldoc.springexample.ampq;
+
+import static com.infosupport.ldoc.springexample.util.StringOperations.stripName;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.infosupport.ldoc.springexample.util.PlantUmlBuilder;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+
+public class SpringAmpqRenderer {
+
+  public static final String RABBIT_LISTENER_ANNOTATION = "org.springframework.amqp.rabbit."
+      + "annotation.RabbitListener";
+  public static final String RABBIT_TEMPLATE_CLASS = "org.springframework.amqp.rabbit.core."
+      + "RabbitTemplate";
```

Comment:
That information is not present in the JSON if I am correct. Or do you mean to both check for the interface as well as the implementation?

Reply:
Exactly, that's what I mean by "treating them the same" here: having the same behavior regardless of whether it is spelled `RabbitTemplate` or `AmqpTemplate` in the source, because the difference doesn't matter for our purposes.

In the Spring Event example renderer, I say that _every_ invocation of _any_ method named `publishEvent` on _any_ class or interface from the Spring Context package counts as a call to `publishEvent`. That means it doesn't matter which actual implementation people use (there are a few, see the All Known Implementing Classes in the [Javadoc](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/ApplicationEventPublisher.html)). You could use a similar approach here if you like.